cpp/src/arrow/compute/api_scalar.h

```diff
@@ -1332,6 +1332,40 @@ Result<Datum> ISOYear(const Datum& values, ExecContext* ctx = NULLPTR);
 ARROW_EXPORT
 Result<Datum> USYear(const Datum& values, ExecContext* ctx = NULLPTR);
 
+/// \brief Year returns PostgreSQL year for each element of `values`
```

Comment:
Here can be a more detailed description of `PGYear` and other types of Year, where are the main differences.